test/test_config.h.in

```diff
@@ -9,16 +9,33 @@
 #define RENDER_ENGINE_VALUES ::testing::ValuesIn(\
     gz::rendering::TestValues())
 
+#include <vector>
+#include <gz/common/Util.hh>
+#include <gz/rendering/config.hh>
+
 /// \todo(anyone) re-enable ogre2 test once ogre 2.2 works on macOS
 #ifdef __APPLE__
 static const std::vector<const char *> kRenderEngineTestValues{"ogre", "optix"};
 #else
-static const std::vector<const char *> kRenderEngineTestValues{"ogre2", "optix"};
+  #if defined(HAVE_OGRE) && defined(HAVE_OPTIX) && defined(HAVE_OGRE2)
+  static const std::vector<const char *> kRenderEngineTestValues{"ogre", "ogre2", "optix"};
```

Comment:
I remember why we have the weird logic with hardcoded engine names. In fortress, we can't run `ogre` and `ogre2` tests one after another due to conflicting symbols in the dynamic library. So if both ogre and ogre2 exists, we prefer running the test with `ogre2`. In this case it would be:

```
static const std::vector<const char *> kRenderEngineTestValues{"ogre2", "optix"};
```

Note that the tests fixture is revamped in https://github.com/gazebosim/gz-rendering/pull/685 so Harmonic does not have this issue any more.